ceramic-one/src/main.rs

```diff
@@ -19,72 +31,231 @@ struct Cli {
+        self.iroh.stop().await?;
+        self.metrics_handle.shutdown();
+        Ok(())
     }
 }
 
-async fn daemon(opts: DaemonOpts) -> Result<()> {
-    let mut metrics_config = MetricsConfig::default();
-    metrics_config = metrics_config_with_compile_time_info(metrics_config);
-    metrics_config.collect = opts.metrics;
-    metrics_config.tracing = opts.tracing;
-    let service_name = metrics_config.service_name.clone();
-    let instance_id = metrics_config.instance_id.clone();
-    let metrics_handle = iroh_metrics::MetricsHandle::new(metrics_config)
-        .await
-        .expect("failed to initialize metrics");
-    info!(service_name, instance_id);
-
-    let dir = match opts.store_dir {
-        Some(dir) => dir,
-        None => match home::home_dir() {
-            Some(home_dir) => home_dir.join(".ceramic-one"),
-            None => PathBuf::from(".ceramic-one"),
-        },
-    };
-    debug!("Using directory: {}", dir.display());
-
-    let store = RocksStoreService::new(dir.join("store")).await?;
-
-    let mut p2p_config = Libp2pConfig::default();
-    // Do not use any bootstrap_peers,
-    // js-ceramic will initialize the bootstrap_peers when it connects.
-    p2p_config.bootstrap_peers = vec![];
-    p2p_config.listening_multiaddrs = vec![
-        "/ip4/0.0.0.0/tcp/0".parse().unwrap(),
-        "/ip4/0.0.0.0/udp/0/quic-v1".parse().unwrap(),
-    ];
-    let p2p = P2pService::new(p2p_config, dir, store.addr()).await?;
-
-    // Note by default this is configured with an indexer, but not with http resolvers.
-    let iroh = IrohBuilder::new().store(store).p2p(p2p).build().await?;
-
-    // Run the HTTP server
-    ceramic_kubo_rpc::http::serve(iroh.api().clone(), opts.bind_address).await?;
-
-    // Stop the system gracefully.
-    iroh.stop().await?;
-
-    metrics_handle.shutdown();
+async fn eye(opts: EyeOpts) -> Result<()> {
+    let daemon = Daemon::build(opts.daemon).await?;
+
+    // Start subscription
+    let subscription = daemon.iroh.api().p2p()?.subscribe(opts.topic).await?;
+
+    let client = daemon.iroh.api().clone();
+    let metrics = daemon.metrics.clone();
+
+    let p2p_events_handle = task::spawn(subscription.for_each(move |event| {
+        match event.expect("should be a message") {
+            iroh_api::GossipsubEvent::Subscribed { .. } => {}
+            iroh_api::GossipsubEvent::Unsubscribed { .. } => {}
+            iroh_api::GossipsubEvent::Message {
+                from,
+                id: _,
+                message,
+            } => {
+                let msg: Message = serde_json::from_slice(message.data.as_slice())
+                    .expect("should be json message");
+                info!(?msg);
+                match &msg {
+                    Message::Update {
+                        stream: _,
+                        tip,
+                        model: _,
+                    } => {
+                        if let Ok(ipfs_path) = IpfsPath::from_str(tip) {
```

Comment:
The `tip` should always be a CID? Why are we using IpfsPath? (I'm not even sure what this function does)

Reply:
Correct the tip is always a CID, however the API to fetch a block expects a path so we create a path that is just the CID.